app/__init__.py

```diff
@@ -35,6 +35,11 @@ def create_app(config_name):
     login_manager.login_view = 'main.render_login'
     main_blueprint.config = application.config.copy()
 
+    @application.before_request
+    def remove_trailing_slash():
+        if request.path.endswith('/'):
+            return redirect(request.path[:-1])
```

Comment:
The Grails app makes URLs like `https://www.digitalmarketplace.service.gov.uk/search?keyword=cloud`. So I think we're safe. Not ideal for the URL to not redirect; is there any easy way to keep the query parameters?